include/state_machine.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "task.h"
+
+namespace graft
+{
+
+class StateMachine final
+{
+public:
+    enum State
+    {
+        EXECUTE,
+        PRE_ACTION,
+        CHK_PRE_ACTION,
+        WORKER_ACTION,
+        CHK_WORKER_ACTION,
+        WORKER_ACTION_DONE,
+        POST_ACTION,
+        CHK_POST_ACTION,
+        AGAIN,
+        EXIT,
+    };
+
+    using St = graft::Status;
+    using Statuses = std::initializer_list<graft::Status>;
+    using Guard = std::function<bool (BaseTaskPtr bt)>;
+    using Action = std::function<void (BaseTaskPtr bt)>;
+
+    StateMachine(State initial_state = EXECUTE)
+    {
+        state(initial_state);
+        init_table();
+    }
+
+    void dispatch(BaseTaskPtr bt, State initial_state)
+    {
+//        state(State(initial_state));
+        state(initial_state);
+        while(state() != EXIT)
+        {
+            process(bt);
+        }
+    }
+
+private:
+    void init_table();
+    State state() const { return m_state; }
+    State state(State state) { return m_state = state; }
+    St status(BaseTaskPtr bt) const { return bt->getLastStatus(); }
+
+    void process(BaseTaskPtr bt)
```

Comment:
Don't we need some logging here??

Reply:
Not sure. On one hand the messages would accompany others logging messages. On the other, they would be useful if we forgot to log such others, and they should be of special categorie, which we have no yet, to not overwhelm the log.

Reply:
please make this PR vs. development/v0.0.2